### PR TITLE
LPS-149354 - fix Zip Slip vulnerability in ContainerLPKGUtil.deploy

### PR DESCRIPTION
**Pull request by jeffwu0724**
Hi Tina, Shuyang,

This is for ticket https://issues.liferay.com/browse/LPS-149354.
Thanks for checking!

### CODE REVIEW

File: modules/apps/static/portal-lpkg-deployer/portal-lpkg-deployer-impl/src/main/java/com/liferay/portal/lpkg/deployer/internal/ContainerLPKGUtil.java [hunk 1]
```diff
@@ -70,13 +72,28 @@ public static List<File> deploy(File lpkgFile, Properties properties)
 					return null;
 				}
 
-				Path lpkgPath = deployerDirPath.resolve(name);
+				File innerLPKGFile = new File(deployerDirPath.toFile(), name);
```
Comment (jeffwu0724) [inline modules/apps/static/portal-lpkg-deployer/portal-lpkg-deployer-impl/src/main/java/com/liferay/portal/lpkg/deployer/internal/ContainerLPKGUtil.java:75]: avoid changing between file, path and string so many times

**Comment by catarineaguiar** [comment]
ci:help

**Comment by catarineaguiar** [comment]
ci:test:lpkg

**Comment by catarineaguiar** [comment]
@jeffwu0724 no regressions found.

File: modules/apps/static/portal-lpkg-deployer/portal-lpkg-deployer-impl/src/test/java/com/liferay/portal/lpkg/deployer/internal/ContainerLPKGUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	@Rule
+	public static final LiferayUnitTestRule liferayUnitTestRule =
+		LiferayUnitTestRule.INSTANCE;
+
+	@Test
+	public void testDeploy() throws Exception {
+		Path targetPath = Paths.get(
+			PropsValues.MODULE_FRAMEWORK_MARKETPLACE_DIR);
+
+		Path testPath = Files.createDirectories(targetPath);
```
Comment (tinatian) [inline modules/apps/static/portal-lpkg-deployer/portal-lpkg-deployer-impl/src/test/java/com/liferay/portal/lpkg/deployer/internal/ContainerLPKGUtilTest.java:62]: @jeffwu0724 is the value of targetPath same as testPath?

**Comment by tinatian** [comment]
@jeffwu0724 Please read your test code one more time, try to simplify it as possible as you can.

Tina

**Pull request closed without merging**